smriprep/utils/misc.py

```diff
@@ -21,3 +21,49 @@ def apply_lut(in_dseg, lut, newpath=None):
                    segm.affine, hdr).to_filename(out_file)
 
     return out_file
+
+
+def fs_isRunning(subjects_dir, subject_id, mtime_tol=86400, logger=None):
+    """
+    Checks FreeSurfer subjects dir for presence of recon-all blocking ``IsRunning`` files,
+    and optionally removes any based on the modification time.
+
+    Parameters
+    ----------
+    subjects_dir : os.PathLike or None
+        Existing FreeSurfer subjects directory
+    subject_id : str
+        Subject label
+    mtime_tol : int
+        Tolerance time (in seconds) between current time and last modification of ``recon-all.log``
+
+    Returns
+    -------
+    subjects_dir : os.PathLike or None
+
+    """
+    from pathlib import Path
+    import time
+
+    if subjects_dir is None:
+        return subjects_dir
+    subj_dir = Path(subjects_dir) / subject_id
+    if not subj_dir.exists():
+        return subjects_dir
+
+    isrunning = tuple(subj_dir.glob("scripts/IsRunning*"))
+    if not isrunning:
+        return subjects_dir
+    reconlog = subj_dir / "scripts" / "recon-all.log"
+    # if recon log doesn't exist, just clear IsRunning
+    mtime = reconlog.stat().st_mtime if reconlog.exists() else 0
+    if (time.time() - mtime) < mtime_tol:
+        raise RuntimeError(f"{subj_dir} contains IsRunning files: {isrunning}\n"
+                           "FreeSurfer will not run if these are present, to avoid "
+                           "interfering with a running process. If no process is running, "
+                           "they may be safely removed.")
```

Comment:
```suggestion
        raise RuntimeError(f"""\
The FreeSurfer's subject folder <{subj_dir}> contains IsRunning files that \
may pertain to a current or past execution: {isrunning}.
FreeSurfer cannot run if these are present, to avoid interfering with a running \
process. Please, make sure no other process is running ``recon-all`` on this subject \
and proceed to delete the files listed above.""")
```